sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuiteWithSpeculation.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.config
+import org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanHelper
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSparkSession
+
+class SQLQuerySuiteWithSpeculation extends QueryTest with SharedSparkSession
```

Comment:
The test doesn't make sense to me. You'd better figure out a unit test?

Reply:
Do you means that this is an integration test and I just need to test the HadoopMapReduceCommitProtocol?

Reply:
> Do you mean that this is an integration test

You can not guarantee a concurrent speculative task in this test, right?

> I just need to test the HadoopMapReduceCommitProtocol?

I think so, if possible.

Reply:
I tried to add an ut for HadoopMapReduceCommitProtocol only, but it seems that this class is an internal class and just used for sql module.

Reply:
Before set SPECULATION_MULTIPLIER to 0, I could not guarantee a concurrent speculative task could be launched.

After set `SPECULATION_MULTIPLIER` to 0 and set `SPECULATION_QUANTILE` to 0.5.
I can ensure that some speculative tasks would be launched.